service/src/main/java/uk/gov/hmcts/reform/fpl/service/RepresentativeService.java

```diff
@@ -0,0 +1,186 @@
+                validationErrors.add(format("Select how %s wants to get case information", representativeLabel));
+            }
+
+            if (EMAIL.equals(servingPreferences)) {
+                if (isEmpty(representative.getEmail())) {
+                    validationErrors.add(format("Enter an email address for %s", representativeLabel));
+                }
+            }
+
+            if (POST.equals(servingPreferences)) {
+                if (isNull(representative.getAddress()) || isEmpty(representative.getAddress().getPostcode())) {
+                    validationErrors.add(format("Enter a postcode for %s", representativeLabel));
+                }
+                if (isNull(representative.getAddress()) || isEmpty(representative.getAddress().getAddressLine1())) {
+                    validationErrors.add(format("Enter a valid address for %s", representativeLabel));
+                }
+            }
+
+            if (nonNull(role) && RESPONDENT.equals(representative.getRole().getType())) {
+                Optional<Representable> responded = findRespondent(caseData, role.getSequenceNo());
+                if (responded.isEmpty()) {
+                    validationErrors.add(format("Respondent %s represented by %s doesn't exist."
+                            + " Choose a respondent who is associated with this case",
+                        representative.getRole().getSequenceNo() + 1, representativeLabel));
+                }
+            }
+
+            if (nonNull(role) && OTHER.equals(role.getType())) {
+                int otherPersonSeq = representative.getRole().getSequenceNo();
+                Optional<Representable> other = findOther(caseData, otherPersonSeq);
+                if (other.isEmpty()) {
+                    String otherPersonLabel = otherPersonSeq == 0 ? "Person" : "Other person " + otherPersonSeq;
+                    validationErrors.add(format("%s represented by %s doesn't exist."
+                            + " Choose a person who is associated with this case",
+                        otherPersonLabel, representativeLabel));
+                }
+            }
+
+            if (DIGITAL_SERVICE.equals(servingPreferences)) {
+                if (isEmpty(representative.getEmail())) {
+                    validationErrors.add(format("Enter an email address for %s", representativeLabel));
+                } else {
+                    Optional<String> userId = organisationService
+                        .findUserByEmail(authorisation, representative.getEmail());
+
+                    if (userId.isEmpty()) {
+                        validationErrors.add(
+                            format("%s must already have an account with the digital service", representativeLabel));
+                    }
+                }
+            }
+        }
+
+        return validationErrors;
+    }
+
+    public void addRepresentatives(CaseData caseData, Long caseId, String auth) {
+        caseData.getRepresentatives().stream()
```

Comment:
looks like there's no need for `.stream()` here

Reply:
done